src/fswAlgorithms/orbitControl/lambertPlanner/lambertPlanner.h

```diff
@@ -0,0 +1,56 @@
+/*
+ ISC License
+
+ Copyright (c) 2023, Autonomous Vehicle Systems Lab, University of Colorado Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+*/
+
+
+#ifndef LAMBERTPLANNER_H
+#define LAMBERTPLANNER_H
+
+#include "architecture/_GeneralModuleFiles/sys_model.h"
+#include "architecture/msgPayloadDefC/NavTransMsgPayload.h"
+#include "architecture/msgPayloadDefC/LambertProblemMsgPayload.h"
+#include "architecture/utilities/bskLogging.h"
+#include "architecture/messaging/messaging.h"
+#include "architecture/utilities/avsEigenSupport.h"
+#include "architecture/utilities/astroConstants.h"
+#include <vector>
+
+/*! @brief This module creates the LambertProblemMsg to be used for the LambertSolver module
+ */
+class LambertPlanner: public SysModel {
+public:
+    LambertPlanner();
+    ~LambertPlanner();
+
+    void Reset(uint64_t CurrentSimNanos) override;
+    void UpdateState(uint64_t CurrentSimNanos) override;
+
+    ReadFunctor<NavTransMsgPayload> navTransInMsg;                      //!< translational navigation input message
+    Message<LambertProblemMsgPayload> lambertProblemOutMsg;             //!< lambert problem output message
+
+    BSKLogger bskLogger;                                                //!< -- BSK Logging
+
+    std::string solverName;         //!< name of lambert algorithm
+    Eigen::Vector3d r_TN_N;         //!< [m] targeted position vector with respect to celestial body at finalTime, in N frame
+    double finalTime{};             //!< [s] time at which target position should be reached
+    double maneuverTime{};          //!< [s] time at which maneuver should be executed
+    double mu{};                    //!< [m^3 s^-2] gravitational parameter
+    int numRevolutions = 0;         //!< [-] number of revolutions to be completed (completed orbits)
+};
+
+#endif
```

Comment:
Similarly I think you could squash this commit into the previous one containing all the main functionality additions

Reply:
Left it the way it was, per our private conversation